### PR TITLE
CATROID-734 Update Design for Terms of Use and Service

### PR DESCRIPTION
**Pull request by dandrawes**
https://jira.catrob.at/browse/CATROID-734
New design for Terms of Use and Service pop-up

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [x] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.java [hunk 1]
```diff
@@ -87,28 +89,54 @@ protected void onCreate(Bundle savedInstanceState) {
...
 						.getInt(AGREED_TO_PRIVACY_POLICY_VERSION, 0);
-		int currentPrivacyPolicyHash = getResources().getString(R.string.dialog_privacy_policy_text)
-						.hashCode();
-		if (oldPrivacyPolicyHash == currentPrivacyPolicyHash) {
+
+		if (oldPrivacyPolicy == Constants.CATROBAT_TERMS_OF_USE_ACCEPTED) {
 			loadContent();
 		} else {
-			setContentView(R.layout.privacy_policy_view);
+			View view = View.inflate(this, R.layout.privacy_policy_view, null);
```
Comment (mstoeg) [inline catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.java:98]: Please outsource this to an extra method (e.g. showTermsOfUseDialog()) in order to keep the onCreate() method compact.

File: catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.java [hunk 2]
```diff
@@ -87,28 +89,54 @@ protected void onCreate(Bundle savedInstanceState) {
...
+
+			String termsOfUseUrl = getString(R.string.terms_of_use_link_template, Constants.CATROBAT_TERMS_OF_USE_URL
+							+ Constants.CATROBAT_TERMS_OF_USE_TOKEN_FLAVOR_URL + BuildConfig.FLAVOR
+							+ Constants.CATROBAT_TERMS_OF_USE_TOKEN_VERSION_URL + BuildConfig.VERSION_CODE,
+							termsOfUseUrlStringText);
+
+			termsOfUseUrlTextView.setText(Html.fromHtml(termsOfUseUrl));
+
+			new AlertDialog.Builder(this)
+					.setNegativeButton(R.string.decline, new DialogInterface.OnClickListener() {
```
Comment (mstoeg) [inline catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.java:114]: Please use lambda-expressions here. They should be suggested by Android Studio automatically.

**Review by mstoeg — CHANGES_REQUESTED** [review]
Looks good! Just have a look at my coding style advice.

**Review by mstoeg — APPROVED** [review]
Well done!

**Review by wslany — CHANGES_REQUESTED** [review]
Thank you, very nice. 

However, when one declines the "Terms of Use and Service" dialog, the following screen appears:
image
Please change the text and name in https://github.com/Catrobat/Catroid/blob/develop/catroid/src/main/res/values/strings.xml#L322 from
```
    <string name="declined_privacy_agreement_alert">To use our app, it is necessary to accept our Privacy Policy.
```
to:
```
    <string name="declined_terms_of_use_and_service_alert">To use our app, it is necessary to accept our Terms of Use and Service.
```
This also has implications for https://github.com/Catrobat/Catroid/blob/develop/catroid/src/main/res/layout/declined_privacy_agreement_alert_view.xml , both in used resources as well as the name of that file, since it does not refer to the privacy policy anymore but instead now to the terms of use and service.

Note that the share.catrob.at link on this second screen does not work. Instead, tapping on the link shall open the community site in the user's preferred browser (_not_ in a webview), much like in the following dialog:
image
Please use the BASE_APP_URL_HTTPS constant from https://github.com/Catrobat/Catroid/blob/5ae47a16b7089a9bcc878f3e82aded57dcd93e60/catroid/src/main/java/org/catrobat/catroid/common/Constants.java#L117 for this link. This ensures that the correct flavor of share is opened here.

Also, when the user then taps OK on that second screen, he or she is shown an empty black screen on which there is no button nor text (tapping on the Android back button apparently exits Catroid). Instead, tapping on OK shall return to the first dialog (which you have correctly implemented).

**Review by mstoeg — CHANGES_REQUESTED** [review]
Looks good now, as far as the remarks from @wslany are concerned. However, there is still an odd behavior if the back button is pressed on the first welcome-dialog. I think instead of showing a black screen it would be better to just finish the activity there.

**Review by mstoeg — APPROVED** [review]

**Review by wslany — APPROVED** [review]

**Comment by wslany** [comment]
Thanks!

**Pull request merged**